src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy

```diff
@@ -104,12 +130,26 @@ class IqPolicyEvaluatorWorkflowStep
     }
 
     @Override
-    FormValidation doCheckIqApplication(@QueryParameter String value) {
+    FormValidation doCheckListAppId(@QueryParameter String value) {
       FormValidation.validateRequired(value)
     }
 
     @Override
-    ListBoxModel doFillIqApplicationItems(@QueryParameter @Nullable String jobCredentialsId, @AncestorInPath Job job) {
+    FormValidation doCheckManualAppId(@QueryParameter String value, @QueryParameter String jobCredentialsId,
+                                      @AncestorInPath Job job)
+    {
+      FormValidation val = FormValidation.validateRequired(value)
+      if (FormValidation.ok() == val) {
+        if (!IqUtil.
+            verifyOrCreateApplication(NxiqConfiguration.serverUrl.toString(), jobCredentialsId, job, value)) {
+          FormValidation.error(Messages._IqPolicyEvaluation_ManualApplicationVerificationFailed())
+        }
+      }
+      return val
```

Comment:
Conversely, here I don't think you _need_ the `return` statement as simply having `val` as the last expression will mean that the value is returned (thanks to Groovy). That's what they're relying on in the other methods in this class.

On the other hand, _I_ do think it's cleaner/more evident to have an explicit return, so I don't have a problem with it; I'm just not sure if there's a particular coding style that we care about for this particular part of our codebase.

Reply:
> On the other hand, I do think it's cleaner/more evident to have an explicit return

+1 to explicit return